### DISCUSSION TITLE
how can an lambda_http example be re-written using lambda_runtime?

### DISCUSSION THREAD

**Discussion by evbo** [aws/aws-lambda-rust-runtime#836] [Q&A]
For consistency in my codebase, I'm wondering how easy would it be to rewrite this http example using the `lambda_runtime` in place of the `lambda_http`?:

https://github.com/awslabs/aws-lambda-rust-runtime/blob/main/examples/http-dynamodb/src/main.rs#L56

One reason for doing this is I've noticed `lambda_runtime` doesn't complain about `Send` missing when instantiating a (dynamodb) `Client` inside the `function_handler`, which I prefer doing.

**Comment by calavera** [comment]
> I'm wondering how easy would it be to rewrite this http example using the lambda_runtime

It's not hard. But we're not going to change the example in this repo.

You can do it yourself by changing `event: Request` in the handler's signature with `event: LambdaEvent<THE_EVENT_STRUCT_THAT_YOU_WANT_TO_RECEIVE>`, and start the function with `lambda_runtime::run`. The location of the request body depends on the struct that the function receives. You'll also need to change the `Response<Body>` that the function returns with an appropriate struct that the service triggering the Lambda function recognizes.

> instantiating a (dynamodb) `Client` inside the `function_handler`, which I prefer doing.

this will initialize the dynamodb client in every invocation, which can potentially add a few hundred milliseconds of latency to every invocation. We don't recommend it.

**Reply by evbo** [reply]
Thank you for this! However, you said "discourage" but are you aware it's currently not possible to instantiate any sort of `Client` inside the function handler for `lambda_http`?

Sometime recently it stopped working, since `Send` is not supported for any `Client`... Does that mean `lambda_runtime` will soon not support instantiating them in the handler either? Or could this be a bug?

**Reply by calavera** [reply]
I don't think there have been any change in this repo for that to stop working recently. I have not looked in the SDK to see if there are changes regarding Send bounds.
